extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageStreamReader.java

```diff
@@ -193,6 +219,11 @@ public BarrageMessage safelyParseFrom(final StreamReaderOptions options,
                         for (int ci = 0; ci < msg.addColumnData.length; ++ci) {
                             msg.addColumnData[ci].data = ChunkInputStreamGenerator.extractChunkFromInputStream(options,
                                     columnChunkTypes[ci], columnTypes[ci], fieldNodeIter, bufferInfoIter, ois);
+
+                            // test for zero-data columns and remove from snapshot columns bitset
+                            if (msg.addColumnData[ci].data.size() == 0 && msg.snapshotColumns != null) {
```

Comment:
Does it matter if there are `0` rows?

Reply:
I have tested this using options that return `0` rows and the correct behavior is obtained.

Reply:
OBE by refactoring `StreamReader` to include the expected columns